### PR TITLE
Introduce feature gates

### PR DESCRIPTION
**Pull request by mareklibra**
First usage: enable/disable Single Node OpenShift

Related PRs:
- https://github.com/openshift-assisted/assisted-ui/pull/395
- OCM 2117

### CODE REVIEW

File: src/components/ui/ClusterEventsToolbar.tsx [hunk 1]
```diff
@@ -121,7 +121,7 @@ const ClusterEventsToolbar: React.FC<ClustersListToolbarProps> = ({
     });
 
     let nextFilters = setNextSelectedValues(type, isChecked, value, filters);
-    nextFilters = setNextSelectAllValue(cluster.hosts?.length ?? 0, nextFilters);
+    nextFilters = setNextSelectAllValue(cluster.hosts?.length || 0, nextFilters);
```
Comment (mareklibra) [inline src/components/ui/ClusterEventsToolbar.tsx:124]: Our development build does not transpile this operator and OCM is not ready for it.

Comment (irosenzw) [reply src/components/ui/ClusterEventsToolbar.tsx:124]: +1. 
bumped into that too which working against uhc-portal

File: src/features/featureGate.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+
+// Must conform Unleash constants
+type AssistedInstallerFeatureType = 'ASSISTED_INSTALLER_SNO_FEATURE';
```
Comment (irosenzw) [inline src/features/featureGate.tsx:4]: nit: do we need the `ASSISTED_INSTALLER` prefix ? 
can we just call it `SNO_FEATURE` or if we do need that prefix: `AI_SNO_FEATURE`

File: src/features/featureGate.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+export const FeatureGateContext = React.createContext<FeatureGateContextType>({
+  isFeatureEnabled: () => false,
+});
+
+export const FeatureGateContextProvider: React.FC<{
+  features: FeatureListType;
+}> = ({ features, children }) => {
+  const isFeatureEnabled = (feature: AssistedInstallerFeatureType) => {
+    // return feature === 'ASSISTED_INSTALLER_SNO_FEATURE' && window.location.hostname.includes('qaprodauth.');
```
Comment (irosenzw) [inline src/features/featureGate.tsx:31]: leftover ?

File: src/features/featureGate.tsx (hunk 2 above)
Comment (mareklibra) [reply src/features/featureGate.tsx:4]: The same constant is added to OCM as well. I would keep the names same to simplify `git grep` by future maintainers.

File: src/features/featureGate.tsx (hunk 3 above)
Comment (mareklibra) [reply src/features/featureGate.tsx:31]: thanks, removed

File: src/components/clusters/NewClusterPage.tsx [hunk 4]
```diff
@@ -44,6 +48,7 @@ type NewClusterFormProps = {
 
 const NewClusterForm: React.FC<NewClusterFormProps> = ({ pullSecret = '', versions }) => {
   const { addAlert, clearAlerts } = React.useContext(AlertsContext);
+  const { isFeatureEnabled } = React.useContext(FeatureGateContext);
```
Comment (jtomasek) [inline src/components/clusters/NewClusterPage.tsx:51]:
```
const isSingleNodeEnabled = useFeature('ASSISTED_INSTALLER_SNO_FEATURE')
```
would be a nicer API.

this hook could be exported from FeatureGateContext module. See HostDialogsContext.tsx for simmilar solution.

**Review by jtomasek — CHANGES_REQUESTED** [review]

**Review by jtomasek — APPROVED** [review]

**Review by jtomasek — APPROVED** [review]

**Pull request merged**